src/convertToObject.js

```diff
@@ -6,7 +6,13 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const strings = sourceString
+    .split(';')
+    .filter((string) => string.includes(':'))
+    .map((string) => string.split(':'))
+    .map(([name, value]) => ({ [name.trim()]: value.trim() }));
+
+  return Object.assign({}, ...strings);
```

Comment:
[CODE KNOWLEDGE] - Don't use iteration methods to modify an already existing array/object. These methods return a new array, so you should use it. In this case, you are using Object.assign to merge objects, which is fine, but be aware that using spread syntax directly within the object literal `{...strings}` would be more modern and concise.